### DISCUSSION TITLE
Is lower RAM usage possible?

### DISCUSSION THREAD

**Discussion by ionGL** [Rem0o/FanControl.Releases#2327] [General]
All LHM sensors Off, no plugins, only NvApiWrapper.

![fancontrol2](https://github.com/Rem0o/FanControl.Releases/assets/80151979/8457a1cd-5001-44dc-aaa3-823f9ab7d35a)

I control just the gpu with a mix of 2 curves.

![fancontrol1](https://github.com/Rem0o/FanControl.Releases/assets/80151979/f38875fd-0aac-4381-a5be-325546000f05)

Start minimized, notify for update at startup off, no temp tray icons.

I saw 29MB ram usage after a boot.

![fancontrol ram1](https://github.com/Rem0o/FanControl.Releases/assets/80151979/dd359c59-d080-488e-956b-d811558c26c1)

But after you open the main window and close it to tray, it remains at ~100MB.

![fancontrol ram2](https://github.com/Rem0o/FanControl.Releases/assets/80151979/269bce67-34f6-472c-820f-8a1231da8355)

Is this a .net thing, not returning to that lower ram usage when the main window was not opened yet? I tried all .net verions and it makes little difference.

I know ram is cheap, but i was wondering if ram usage could be lower in a case like this, when LHM is not needed.

**Comment by Rem0o** [comment]
If you wait for some time, forget about it, then open Task manager back, you'll see it will go down if you leave the UI closed.   That's garbage collection. I don't have complete control over it.  Might take 5 minutes, might take 30 minutes, but it will go down eventually.

**Reply by Rem0o** [reply]
Like this:
![image](https://github.com/Rem0o/FanControl.Releases/assets/14118956/b98ac626-cc70-4e9f-8d1a-419e2d52a723)

**Reply by ionGL** [reply]
I kept an eye on the 4.8 version since yesterday. Windows restart, waited for it to start in tray, opened it for a few seconds, then back to tray an left it there. Checked task manger from time to time, it didn't move lower than ~85MB.

I'm now testing with 8.0. The thing is i rarely restart windows. I just put it on standby. Once it resumes from standby, i can see in task manager that it jumped to ~136MB. I don't touch it, tested with a 3d load and the gpu curves work like they should. It stays at ~136 and only goes down once the main window is opened, goes to ~98, then in tray stays around that. 

I notice that if you open it from tray after a standby, you see the "configuration loaded..." notification. 

If you resume from standby and never touch it, it will stay ~136 forever.

**Reply by ionGL** [reply]
I haven't touched the tray icon since yesterday, no windows restart, a couple standbys, it's not moving lower than 139. Gpu fan control works.

![fancontrolram1day](https://github.com/Rem0o/FanControl.Releases/assets/80151979/7115f983-b4f8-4597-a887-972474e3b9e4)

**Reply by ionGL** [reply]
After another 2 days of leaving it alone in tray, no windows restarts, around 10-15 standbys, it stabilized at ~141MB.

Then when i opened it from tray after a few days, i got the "configuration loaded" notification in the interface many times one after the other. Probably it kept count from the standbys and now it reloaded them all.

Edit: I tried again 2 standbys. Once opened, got 2 "configuration loaded" messages.

**Reply by Rem0o** [reply]
The configloaded message is normal. There is a queue and it waits for the window to come back to show the messages.

**Comment by ionGL** [comment]
Writing after the 2.11 update. Ram usage is now much lower. It goes to 15MB as soon as i close it to tray and goes to 40-50MB after a minute. The good thing is it stays there if you leave it alone, doesn't go up after standbys.
